### ISSUE TITLE
Unable to map Active Directory service account to Hashicorp vault AD secrets engine

### ISSUE THREAD

**Issue by ahladh** [hashicorp/vault#6110]
When I enable, add AD configuration and I try to map a service account in Active Directory I get error saying "Network ERROR".

**$**  vault write ad/config     binddn=$USERNAME     bindpass=$PASSWORD     url=ldaps://ahladh.tk     userdn='dc=ahladh,dc=tk'
Success! Data written to: ad/config

**$** vault write ad/roles/poc.test    service_account_name="poc.test@ahladh.tk"

**Error writing data to ad/roles/poc.test: Error making API request.

URL: PUT http://52.4.126.19:8200/v1/ad/roles/poc.test
Code: 500. Errors:

* 1 error occurred:
        * error connecting to host "ldaps://ahladh.tk": LDAP Result Code 200 "Network Error": read tcp 10.0.1.15:52986->35.18.1.2:636: read: connection reset by peer**

I have an SSL certificate installed already in the Active Directory
Could Someone crack this . Thanks in advance.

**Comment by raygj** [comment]
I ran into this issue as well and resolved it by verifying the full account name I had provisioned for Vault to access AD. In my case I was using the short name "vault" rather than the full name "vault service." I found this by using the dsquery command `dsquery user -name v*` on a domain controller which returned the fully qualified LDAP string: "CN=Vault Service,CN=Users,DC=ad,DC=lab,DC=org". I replaced "vault" with "vault service" in the Vault configuration stanza (item 2 [HERE](https://www.vaultproject.io/docs/secrets/ad/index.html)) used to create accounts in AD.
